subsys/net/lib/lwm2m/lwm2m_obj_firmware.c

```diff
@@ -113,67 +126,80 @@ void lwm2m_firmware_set_update_state(uint8_t state)
 
 	if (error) {
 		LOG_ERR("Invalid state transition: %u -> %u",
-			update_state, state);
+			update_state[obj_inst_id], state);
 	}
 
-	update_state = state;
-	NOTIFY_OBSERVER(LWM2M_OBJECT_FIRMWARE_ID, 0, FIRMWARE_STATE_ID);
-	LOG_DBG("Update state = %d", update_state);
+	snprintf(path, sizeof(path), "5/%" PRIu16 "/3", obj_inst_id);
+
+	lwm2m_engine_set_u8(path, state);
+
+	LOG_DBG("Update state = %d", state);
+}
+
+void lwm2m_firmware_set_update_state(uint8_t state)
+{
+	lwm2m_firmware_set_update_state_inst(0, state);
+}
+
+uint8_t lwm2m_firmware_get_update_result_inst(uint16_t obj_inst_id)
+{
+	return update_result[obj_inst_id];
 }
 
 uint8_t lwm2m_firmware_get_update_result(void)
```

Comment:
This function should be removed as it's not needed anymore.

Reply:
It is needed for the backwards compatibility